store/v2/storage/sqlite/db.go

```diff
@@ -2,13 +2,13 @@ package sqlite
 
 import (
 	"bytes"
-	"database/sql"
 	"errors"
 	"fmt"
 	"path/filepath"
 	"strings"
+	"sync"
 
-	_ "github.com/mattn/go-sqlite3"
```

Comment:
what is the reason for the switch?

Reply:
CGO doesn't appear to support the use case of two separate go libraries wrapping the same C lib, I got errors like below at build time.
```
duplicate symbol '_sqlite3_vsnprintf' in:
    /private/var/folders/94/t_3t6kq94s16r63_kc66qxxr0000gn/T/go-link-79727755/000028.o
    /private/var/folders/94/t_3t6kq94s16r63_kc66qxxr0000gn/T/go-link-79727755/000037.o
```
So I reasoned we should use one SQLite library in both iavl/v2 and store/v2.  iavl/v2 uses `bvinc/go-sqlite-lite` which is lower level and doesn't satisfy `database/sql`; it felt harder to shift iavl/v2 to `mattn` than vice versa.

The SS sqlite code changed a lot, we could also try a pure go lib like `modernc` which *does* satisfy `database/sql` so most of the code would be the same.

Reply:
i think we left the modernc lib as it wasnt as performant as the cgo version. wanted to better understand the change but looks good